extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/sql/HllSketchBaseSqlAggregator.java

```diff
@@ -154,19 +160,23 @@ public Aggregation toDruidAggregation(
       }
 
       if (inputType.is(ValueType.COMPLEX)) {
-        aggregatorFactory = new HllSketchMergeAggregatorFactory(
-            aggregatorName,
-            dimensionSpec.getOutputName(),
-            logK,
-            tgtHllType,
-
-            // For HllSketchMergeAggregatorFactory, stringEncoding is only advisory to aid in detection of mismatched
-            // merges. It does not affect the results of the aggregator. At this point in the code, we do not know what
-            // the input encoding of the original sketches was, so we set it to the default.
-            HllSketchAggregatorFactory.DEFAULT_STRING_ENCODING,
-            finalizeSketch || SketchQueryContext.isFinalizeOuterSketches(plannerContext),
-            ROUND
-        );
+        if (HllSketchMergeAggregatorFactory.TYPE.equals(inputType) ||
+            Objects.equals(inputType.getComplexTypeName(), "HLLSketch") ||
+            Objects.equals(inputType.getComplexTypeName(), "HLLSketchBuild")) {
```

Comment:
Let's refrain from using string variables here and replace them with already present static strings.